### ISSUE TITLE
Banded columns and virtual table bug

### ISSUE THREAD

**Issue by imaxer** [DevExpress/devextreme-reactive#1389]
I use VirtualTable and BandedColumns together, column header cells behave strangely when horizontal scrolling. I think colspan is calculated incorrectly for banded columns. When the first column from group is removed from DOM, its banded header cell it is also removed

- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...

- [x] React Grid
- [ ] React Chart
- [ ] Vue Grid

**Comment by kvet** [comment]
https://codesandbox.io/s/6xw9q1j2pz

**Comment by ushkal** [comment]
We have thoroughly researched the issue and decided to fix it in the **next major release** because our preliminary solution requires a breaking change. 
Let me explain the issue in greater detail: when both virtual table and banded columns are enabled, the virtual table layout depends on the table header rows that the TableBandHeader plugin provides. On the other hand, in order to correctly calculate colspan for columns, the TableBandHeader plugin needs to know the visible table boundaries that the Virtual Table provides. 
To resolve this circular dependency we are going to extract a virtualization logic to a standalone plugin named "VirtualScrolling" that will be used in conjunction with the Table plugin:

before: 
```
<Grid>
  <VirtualTable />
  ...
  <TableBandHeader />
  ...
</Grid>
```
 
after:
```
<Grid>
  <Table />
  ...
  <TableBandHeader />
  ...
  <VirtualScrolling />
</Grid> 
```

**Comment by Hatish** [comment]
I have run into this issue as well. Will this fix be included in version 1.11 (currently alpha1)?

**Comment by vamsi0212** [comment]
@ushkal  Any update on this fix?

**Comment by ushkal** [comment]
Hi @vamsi0212,

We're still working on this issue. We'll update the thread once we the problem is fixed.

**Comment by ccamusso** [comment]
Do you have workarounds to suggest in the meantime?

**Comment by pfronczak** [comment]
This is the workaround I created, when I hit this issue:
https://codesandbox.io/s/virtualtable-banded-columns-workaround-ckzjv

It's a custom plugin that adds the band title as an absolutely positioned element on top of the header cell.
It's not perfect - it only supports a single level of bands, the borders are only partially supported, but it was good enough for our use case.

It works with the same format of columnBands property and also works fine when combined with `TableColumnVisibility` plugin.

Any feedback welcome :)

**Comment by ccamusso** [comment]
It works but the TH style is too generic.
With two grids, one with a column band and other without a column band, If I change the style of the TH, I create a little style problem on the second grid.

**Comment by pfronczak** [comment]
Absolutely - the TH style is there only to keep the example shorter. In reality it would be better to either override the header row component to introduce some styling or to add some scoping class for the whole table and use that to only style the relevant cells.

**Comment by IndigoHollow** [comment]
Hello. Any updates on this fix?

**Comment by ushkal** [comment]
Hi @IndigoHollow ,
 
We are currently working on a solution that would not require a breaking change. We'll update this thread once this issue is fixed.

**Comment by IndigoHollow** [comment]
Inside of issue #2139 is written that it awaits when this issue will be fixed. Issue #2139 was closed recently. Could you please kindly give an update on this current issue?

**Comment by ushkal** [comment]
Hi @IndigoHollow ,
 
We've found another way to fix issue #2139 . The fix for the current issue is almost ready. We expect to merge it next week.
